client/src/elm/Pages/MessagingCenter/View.elm

```diff
@@ -70,36 +66,64 @@ view language currentTime nurseId nurse db model =
                     let
                         messagingCenterView =
                             viewMessagingCenter language currentTime currentDate programStartDate nurseId nurse db model
+
+                        surveys =
+                            Dict.get nurseId db.resilienceSurveysByNurse
+                                |> Maybe.andThen RemoteData.toMaybe
+                                |> Maybe.map Dict.values
+                                |> Maybe.withDefault []
+
+                        surveysSorted =
+                            List.sortWith (sortByDateDesc .dateMeasured) surveys
+
+                        surveyCount surveyType =
```

Comment:
This is not necessary.
Filtering should be done once, inside ` runSurvey surveyType`.